recipes/lief/all/conanfile.py

```diff
@@ -79,11 +69,16 @@ def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("mbedtls/3.2.1")
+        # TODO: Check if we actually need transitive_headers
+        # LIEF.lib(x509.obj) external symbol mbedtls_snprintf referenced in function lief_mbedtls_x509_dn_gets
+        self.requires("mbedtls/3.2.1", transitive_headers=True, transitive_libs=True)
```

Comment:
@AbrilRBS - this is not a case of transitive_libs (not even transitive_headers), as the library is already added in the linker statement by conan when creating a static library.

```
1>  LIEF.lib(x509.obj) : error LNK2019: unresolved external symbol mbedtls_snprintf referenced in function "int __cdecl `anonymous namespace'::lief_mbedtls_x509_dn_gets(char *,unsigned __int64,struct mbedtls_asn1_named_data const *)" (?lief_mbedtls_x509_dn_gets@?A0x25ca5665@@YAHPEAD_KPEBUmbedtls_asn1_named_data@@@Z)
1>  conan-center-index\recipes\lief\all\test_package\build\msvc-194-x86_64-17-release\Release\test_package.exe : fatal error LNK1120: 1 unresolved externals
1>Done building target "Link" in project "test_package.vcxproj" -- FAILED.
```

The implementation of lief, is calling into `mbedtls_snprintf` which _should_ be in `mbedx509.lib` - which is linked in the test package that is failing.

The problem is the logic inside `platform.h` in mbedtls - that function is defined to be one thing or the other, depending on the macro deductions in that file - so this impacts how mbedtls is built, and what value the macros resolve to when mbedtls is consumed - and presumably they are different at those two points, that's why the "wrong" symbol is requested

the culprit is in the mbedtls recipe:
https://github.com/luzpaz/conan-center-index/blob/6ed97106162ee022c6f0db0db4478bd18bca5227/recipes/mbedtls/all/conanfile.py#L90-L94

Which is puzzling as that should not be needed if the header does the autodetuction by itself, and it is not mirroring anything that the upstream CMake does (the generated cmake files also dont propagate any compiler definitions). Perhaps you have more insights since you were involved in approving that PR - I may be missing something.